liquigraph-examples/dagger2/src/test/java/org/liquigraph/examples/dagger2/Neo4jVersionReader.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2014-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.examples.dagger2;
+
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+class Neo4jVersionReader {
+
+    public Neo4jVersion read(String classpathResource) {
+        Properties properties = new Properties();
+        try (InputStream input = this.getClass().getResourceAsStream(classpathResource)) {
+            properties.load(input);
+            return Neo4jVersion.parse(properties.getProperty("neo4j.version"));
+        } catch (IOException e) {
+            throw new RuntimeException(e.getMessage(), e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)